--- /dev/null	2026-10-05 20:22:36.000000000 +0200
+++ gml/gml_GlobalScript_Optimizers.gml	2026-09-24 15:31:46.000000000 +0200
@@ -0,0 +1,143 @@
+function optimizer_unary_expression_simplify(arg0)
+{
+}
+
+function optimizer_binary_expression_simplify(arg0)
+{
+}
+
+function optimizer_ternary_expression_simplify(arg0)
+{
+}
+
+function optimizer_compile_time_variable(arg0)
+{
+}
+
+function optimizer_compile_time_function(arg0)
+{
+}
+
+function optimizer_string_literal_merge(arg0)
+{
+}
+
+function optimizer_dead_branch_remove(arg0)
+{
+}
+
+function optimizer_unreachable_code_remove(arg0)
+{
+}
+
+function optimizer_empty_block_remove(arg0)
+{
+}
+
+function optimizer_redundant_else_remove(arg0)
+{
+}
+
+function optimizer_block_flatten(arg0)
+{
+}
+
+function optimizer_condition_merge(arg0)
+{
+}
+
+function optimizer_branch_invert(arg0)
+{
+}
+
+function optimizer_constant_propagate(arg0)
+{
+}
+
+function optimizer_copy_propagate(arg0)
+{
+}
+
+function optimizer_dead_local_remove(arg0)
+{
+}
+
+function optimizer_dead_assignment_remove(arg0)
+{
+}
+
+function optimizer_duplicate_branch_merge(arg0)
+{
+}
+
+function optimizer_redundant_condition_remove(arg0)
+{
+}
+
+function optimizer_common_tail_merge(arg0)
+{
+}
+
+function optimizer_switch_simplify(arg0)
+{
+}
+
+function optimizer_loop_constant_simplify(arg0)
+{
+}
+
+function optimizer_redundant_type_check_remove(arg0)
+{
+}
+
+function optimizer_loop_invariant_hoist(arg0)
+{
+}
+
+function optimizer_loop_unswitch(arg0)
+{
+}
+
+function optimizer_loop_unroll_small(arg0)
+{
+}
+
+function optimizer_local_cse(arg0)
+{
+}
+
+function optimizer_lookup_cache(arg0)
+{
+}
+
+function optimizer_trivial_function_inline(arg0)
+{
+}
+
+function optimizer_single_use_function_inline(arg0)
+{
+}
+
+function optimizer_declared_function_inline(arg0)
+{
+}
+
+function optimizer_assignment_operator_simplify(arg0)
+{
+}
+
+function optimizer_static_literal_hoist(arg0)
+{
+}
+
+function optimizer_condition_reorder(arg0)
+{
+}
+
+function optimizer_temporary_coalesce(arg0)
+{
+}
+
+function optimizer_minify_format(arg0)
+{
+}
