--- gml_138/gml_GlobalScript_s_draw_gui.gml	2026-09-02 08:45:16.118546900 +0200
+++ gml/gml_GlobalScript_s_draw_gui.gml	2026-10-05 10:01:02.000000000 +0200
@@ -95,6 +95,8 @@
         draw_text(20, 20, "button =" + string(instance_number(obj_button)));
         draw_text(20, 30, "window =" + string(instance_number(obj_window)));
     }
+    
+    scr_gnx_draw_scrollbar();
 }
 
 function scr_draw_raid_button()
