--- /dev/null	2026-10-05 20:22:36.000000000 +0200
+++ gml/gml_GlobalScript_normalize.gml	2026-09-24 15:31:46.000000000 +0200
@@ -0,0 +1,9 @@
+function normalize(arg0)
+{
+    var _magnitude = sqrt((arg0.x * arg0.x) + (arg0.y * arg0.y) + (arg0.z * arg0.z));
+    var _result = new Vector3();
+    _result.x = arg0.x / _magnitude;
+    _result.y = arg0.y / _magnitude;
+    _result.z = arg0.z / _magnitude;
+    return _result;
+}
