--- gml_138/gml_GlobalScript_s_unit_function.gml	2026-09-02 08:45:16.530589000 +0200
+++ gml/gml_GlobalScript_s_unit_function.gml	2026-09-24 15:31:46.000000000 +0200
@@ -18,6 +18,8 @@
     _unit_info.preg_c = _info.preg_c;
     _unit_info.preg_c_max = _info.preg_c;
     _unit_info.scr_unit = _info.scr_unit;
+    _unit_info.moan_voice = _info.moan_voice;
+    _unit_info.moan_pitch = _info.moan_pitch;
     return _unit_info;
 }
 
