--- gml_138/gml_GlobalScript_s_food.gml	2026-09-02 08:45:16.155545700 +0200
+++ gml/gml_GlobalScript_s_food.gml	2026-10-05 10:01:02.000000000 +0200
@@ -80,7 +80,7 @@
                 break;
             
             case UnknownEnum.Value_4:
-                _raid = "Consumo de raciones = ";
+                _raid = "Consumo de comida = ";
                 _consume = "Consumidas al final del día";
                 break;
             
