--- gml_138/gml_Object_obj_menu_Draw_0.gml	2026-09-02 08:45:16.156546900 +0200
+++ gml/gml_Object_obj_menu_Draw_0.gml	2026-09-24 15:31:46.000000000 +0200
@@ -29,4 +29,7 @@
 }
 
 draw_set_halign(fa_right);
-draw_text(479, 263, "v." + string(1.38));
+draw_text(479, 263, "v." + string(1.39));
+
+if (variable_global_exists("gnx_version"))
+    draw_text(479, 253, "GNX " + string(global.gnx_version) + " enabled");
