--- gml_138/gml_GlobalScript_s_button_text.gml	2026-09-02 08:45:16.399590800 +0200
+++ gml/gml_GlobalScript_s_button_text.gml	2026-10-05 10:01:02.000000000 +0200
@@ -6,9 +6,27 @@
     var _code_text = -1;
     var _shop_price = -1;
     
+    if (ds_map_exists(global.cell_registry, arg0))
+    {
+        var _gnx_cell = ds_map_find_value(global.cell_registry, arg0);
+        _button_text = _gnx_cell.name;
+        
+        if (variable_struct_exists(_gnx_cell, "gnx_price"))
+        {
+            _price = _gnx_cell.gnx_price;
+            _spawn_info = _gnx_cell.gnx_spawn_info;
+        }
+        
+        if (variable_struct_exists(_gnx_cell, "gnx_code_text"))
+            _code_text = _gnx_cell.gnx_code_text;
+        
+        if (_price != -1)
+            _shop_price = _price;
+    }
+    
     switch (arg0)
     {
-        case UnknownEnum.Value_1:
+        case 1:
             _button_text = "WALL 1";
             _code_text = "A";
             _price = 50;
@@ -21,7 +39,7 @@
             };
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             _button_text = "WALL 2";
             _code_text = "A";
             _price = 50;
@@ -34,7 +52,7 @@
             };
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             _button_text = "WALL 3";
             _code_text = "B";
             _price = 100;
@@ -48,7 +66,7 @@
             };
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             _button_text = "WALL 4";
             _code_text = "C";
             _price = 150;
@@ -62,7 +80,7 @@
             };
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             _button_text = "M.PRESS 1";
             _code_text = "D";
             _price = 280;
@@ -75,7 +93,7 @@
             };
             break;
         
-        case UnknownEnum.Value_13:
+        case 13:
             _button_text = "PRONE";
             _code_text = "E";
             _price = 320;
@@ -89,7 +107,7 @@
             };
             break;
         
-        case UnknownEnum.Value_14:
+        case 14:
             _button_text = "RIDE 1";
             _code_text = "F";
             _price = 350;
@@ -103,25 +121,25 @@
             };
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             _button_text = "BIRTH 1";
             _code_text = "G";
             _price = 30;
             break;
         
-        case UnknownEnum.Value_12:
+        case 12:
             _button_text = "BIRTH 2";
             _code_text = "G";
             _price = 350;
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             _button_text = "MILK 1";
             _code_text = "H";
             _price = 40;
             break;
         
-        case UnknownEnum.Value_10:
+        case 10:
             _button_text = "DISPLAY";
             _code_text = "H";
             _price = 150;
@@ -135,7 +153,7 @@
             };
             break;
         
-        case UnknownEnum.Value_11:
+        case 11:
             _button_text = "DRINK";
             _code_text = "I";
             _price = 320;
@@ -148,7 +166,7 @@
             };
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             _button_text = "ORAL";
             _code_text = "J";
             _price = 220;
@@ -162,7 +180,7 @@
             };
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             _button_text = "TITFUCK";
             _code_text = "K";
             _price = 200;
@@ -175,7 +193,7 @@
             };
             break;
         
-        case UnknownEnum.Value_15:
+        case 15:
             _button_text = "FOREPLAY";
             _code_text = "L";
             _price = 300;
@@ -189,7 +207,7 @@
             };
             break;
         
-        case UnknownEnum.Value_31:
+        case 31:
             _button_text = "T.WALL 1";
             _code_text = "A";
             _price = 300;
@@ -202,7 +220,7 @@
             };
             break;
         
-        case UnknownEnum.Value_32:
+        case 32:
             _button_text = "T.WALL 2";
             _code_text = "A";
             _price = 300;
@@ -215,7 +233,7 @@
             };
             break;
         
-        case UnknownEnum.Value_29:
+        case 29:
             _button_text = "T.WALL 3";
             _code_text = "B";
             _price = 350;
@@ -229,7 +247,7 @@
             };
             break;
         
-        case UnknownEnum.Value_27:
+        case 27:
             _button_text = "BIND 1";
             _code_text = "C";
             _price = 380;
@@ -242,7 +260,7 @@
             };
             break;
         
-        case UnknownEnum.Value_28:
+        case 28:
             _button_text = "BIND 2";
             _code_text = "D";
             _price = 380;
@@ -255,27 +273,27 @@
             };
             break;
         
-        case UnknownEnum.Value_34:
+        case 34:
             _button_text = "CLEAN";
             _code_text = "E";
             _price = 250;
             break;
         
-        case UnknownEnum.Value_33:
+        case 33:
             _button_text = "RECOVER";
             _code_text = "G";
             _price = 450;
             _shop_price = 340;
             break;
         
-        case UnknownEnum.Value_30:
+        case 30:
             _button_text = "MILK 2";
             _code_text = "H";
             _price = 400;
             _shop_price = 520;
             break;
         
-        case UnknownEnum.Value_19:
+        case 19:
             _button_text = "G.BANG 1";
             _price = 350;
             _code_text = "A";
@@ -289,7 +307,7 @@
             };
             break;
         
-        case UnknownEnum.Value_18:
+        case 18:
             _button_text = "G.BANG 2";
             _price = 450;
             _code_text = "B";
@@ -303,7 +321,7 @@
             };
             break;
         
-        case UnknownEnum.Value_20:
+        case 20:
             _button_text = "M.PRESS 2";
             _price = 580;
             _code_text = "C";
@@ -316,7 +334,7 @@
             };
             break;
         
-        case UnknownEnum.Value_21:
+        case 21:
             _button_text = "BEHIND";
             _price = 620;
             _code_text = "D";
@@ -330,7 +348,7 @@
             };
             break;
         
-        case UnknownEnum.Value_22:
+        case 22:
             _button_text = "RIDE 2";
             _price = 650;
             _code_text = "E";
@@ -343,7 +361,7 @@
             };
             break;
         
-        case UnknownEnum.Value_23:
+        case 23:
             _button_text = "G.BANG 3";
             _price = 650;
             _code_text = "F";
@@ -357,25 +375,25 @@
             };
             break;
         
-        case UnknownEnum.Value_17:
+        case 17:
             _button_text = "TRANSFER";
             _price = 350;
             _code_text = "G";
             break;
         
-        case UnknownEnum.Value_24:
+        case 24:
             _button_text = "PATROL";
             _price = 1000;
             _code_text = "H";
             break;
         
-        case UnknownEnum.Value_25:
+        case 25:
             _button_text = "CLONE";
             _price = 4000;
             _code_text = "I";
             break;
         
-        case UnknownEnum.Value_36:
+        case 36:
             _button_text = "DAIRY";
             _price = 400;
             _code_text = "A";
@@ -389,7 +407,7 @@
             };
             break;
         
-        case UnknownEnum.Value_37:
+        case 37:
             _button_text = "GIANT";
             _price = 800;
             _code_text = "B";
@@ -403,7 +421,7 @@
             };
             break;
         
-        case UnknownEnum.Value_38:
+        case 38:
             _button_text = "CHAINS";
             _price = 1000;
             _code_text = "C";
@@ -417,26 +435,26 @@
             };
             break;
         
-        case UnknownEnum.Value_42:
+        case 42:
             _button_text = "R.SHRINE";
             _price = 700;
             _code_text = "R";
             break;
         
-        case UnknownEnum.Value_41:
+        case 41:
             _button_text = "F.SHRINE";
             _price = 1000;
             _code_text = "F";
             _shop_price = 600;
             break;
         
-        case UnknownEnum.Value_40:
+        case 40:
             _button_text = "S.SHRINE";
             _price = 2000;
             _code_text = "S";
             break;
         
-        case UnknownEnum.Value_39:
+        case 39:
             _button_text = "L.SHRINE";
             _price = 4000;
             _code_text = "L";
@@ -466,28 +484,34 @@
     
     switch (arg0)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             _data.button_text = "CELL";
             _data.price = 100;
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             _data.button_text = "BIG CELL";
             _data.price = 300;
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             _data.button_text = "TENTACLE";
             _data.price = 150;
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             _data.button_text = "ENTRANCE";
             _data.price = 60;
             break;
+        
+        case 4:
+        case 6:
+            _data.button_text = "XL CELL";
+            _data.price = 500;
+            break;
     }
     
-    if (global.val.language != UnknownEnum.Value_0)
+    if (global.val.language != 0 && arg0 != 4 && arg0 != 6)
         _data.button_text = scr_lang_slot_text_2(arg0);
     
     return _data;
@@ -508,10 +532,10 @@
         var _md = scr_calculate_mood_rate();
         _c_num = floor(_c_num * max(1, _md.md_rate));
         global.val.money += _c_num;
-        scr_check_steam_achievement(UnknownEnum.Value_4);
+        scr_check_steam_achievement(4);
         var _x = x + (image_xscale * 0.5);
         var _y = y - 50;
-        scr_icon_pop_up_create(_x, _y, _c_num, UnknownEnum.Value_2, false);
+        scr_icon_pop_up_create(_x, _y, _c_num, 2, false);
         scr_play_get_coin_sfx(_x, _y);
     }
     
@@ -525,7 +549,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (button_num)
             {
                 case 0:
@@ -547,7 +571,7 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (button_num)
             {
                 case 0:
@@ -569,7 +593,7 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (button_num)
             {
                 case 0:
@@ -591,7 +615,7 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (button_num)
             {
                 case 0:
@@ -613,7 +637,7 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (button_num)
             {
                 case 0:
@@ -635,7 +659,7 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (button_num)
             {
                 case 0:
@@ -657,7 +681,7 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (button_num)
             {
                 case 0:
@@ -679,7 +703,7 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (button_num)
             {
                 case 0:
@@ -701,7 +725,7 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (button_num)
             {
                 case 0:
@@ -723,7 +747,7 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (button_num)
             {
                 case 0:
@@ -762,7 +786,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (button_num)
             {
                 case 0:
@@ -788,7 +812,7 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (button_num)
             {
                 case 0:
@@ -814,7 +838,7 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (button_num)
             {
                 case 0:
@@ -840,7 +864,7 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (button_num)
             {
                 case 0:
@@ -866,7 +890,7 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (button_num)
             {
                 case 0:
@@ -892,7 +916,7 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (button_num)
             {
                 case 0:
@@ -918,7 +942,7 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (button_num)
             {
                 case 0:
@@ -944,7 +968,7 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (button_num)
             {
                 case 0:
@@ -970,7 +994,7 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (button_num)
             {
                 case 0:
@@ -996,7 +1020,7 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (button_num)
             {
                 case 0:
@@ -1029,7 +1053,7 @@
     scr_create_button_slot(interact_type, button_num);
     button_text = scr_set_prop_text(button_num).text_type;
     
-    if (global.val.language != UnknownEnum.Value_0)
+    if (global.val.language != 0 && button_num < 100)
         button_text = scr_lang_prop_text(button_num);
 }
 
@@ -1047,30 +1071,30 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "PILLAR";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "PLANK";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "ROCK";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "TENDRILS";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "BRICK";
                     _code_text = "E";
                     break;
@@ -1078,30 +1102,30 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "門柱";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "木板";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "巖石";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "卷鬚";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "磚";
                     _code_text = "E";
                     break;
@@ -1109,30 +1133,30 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "门柱";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "木板";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "岩石";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "卷须";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "砖";
                     _code_text = "E";
                     break;
@@ -1140,30 +1164,30 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "柱";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "板";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "岩";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "蔓";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "レンガ";
                     _code_text = "E";
                     break;
@@ -1171,30 +1195,30 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "기둥";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "판자";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "돌";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "촉수";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "벽돌";
                     _code_text = "E";
                     break;
@@ -1202,30 +1226,30 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "КОЛОННА";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "ДОСКА";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "КАМЕНЬ";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "ЩУПАЛЬЦА";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "КИРПИЧ";
                     _code_text = "E";
                     break;
@@ -1233,30 +1257,30 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "PILAR";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "TABLA";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "ROCA";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "TENTÁCULO";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "LADRILLO";
                     _code_text = "E";
                     break;
@@ -1264,30 +1288,30 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "PILIER";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "PLANCHE";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "ROCHE";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "TENTACULES";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "BRIQUE";
                     _code_text = "E";
                     break;
@@ -1295,30 +1319,30 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "SÄULE";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "PLANKE";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "STEIN";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "RANKEN";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "ZIEGEL";
                     _code_text = "E";
                     break;
@@ -1326,30 +1350,30 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (arg0)
             {
-                case UnknownEnum.Value_1:
+                case 1:
                     _name = "PILAR";
                     _code_text = "A";
                     break;
                 
-                case UnknownEnum.Value_0:
+                case 0:
                     _name = "TÁBUA";
                     _code_text = "B";
                     break;
                 
-                case UnknownEnum.Value_2:
+                case 2:
                     _name = "PEDRA";
                     _code_text = "C";
                     break;
                 
-                case UnknownEnum.Value_4:
+                case 4:
                     _name = "FILAMENTOS";
                     _code_text = "D";
                     break;
                 
-                case UnknownEnum.Value_3:
+                case 3:
                     _name = "TIJOLO";
                     _code_text = "E";
                     break;
@@ -1367,219 +1391,233 @@
 
 function scr_set_prop_text(arg0)
 {
+    if (arg0 >= 100)
+    {
+        var _rprop = ds_map_find_value(global.prop_registry, arg0);
+        
+        if (_rprop != undefined)
+        {
+            _data = {};
+            _data.text_type = _rprop.display_name_key;
+            _data.code_text = _rprop.code_text;
+            _data.shop_price = _rprop.price;
+            return _data;
+        }
+    }
+    
     var _code_text = -1;
     var _name = -1;
     var _shop_price = -1;
     
     switch (arg0)
     {
-        case UnknownEnum.Value_37:
+        case 37:
             _name = "FIRE";
             _code_text = "A";
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             _name = "HELM";
             _code_text = "A";
             break;
         
-        case UnknownEnum.Value_0:
+        case 0:
             _name = "CRATE 1";
             _code_text = "A";
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             _name = "CRATE 2";
             _code_text = "C";
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             _name = "FENCE";
             _code_text = "D";
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             _name = "BARREL";
             _code_text = "E";
             _shop_price = 220;
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             _name = "BOTTLE 1";
             _code_text = "F";
             _shop_price = 120;
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             _name = "BOTTLE 2";
             _code_text = "G";
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             _name = "BOARD 1";
             _code_text = "H";
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             _name = "SHIELD";
             _code_text = "I";
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             _name = "SWORD";
             _code_text = "J";
             break;
         
-        case UnknownEnum.Value_11:
+        case 11:
             _name = "BONE 1";
             _code_text = "K";
             _shop_price = 200;
             break;
         
-        case UnknownEnum.Value_12:
+        case 12:
             _name = "BONE 2";
             _code_text = "L";
             break;
         
-        case UnknownEnum.Value_13:
+        case 13:
             _name = "BONE 3";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_10:
+        case 10:
             _name = "ROPE";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_14:
+        case 14:
             _name = "BOARD 2";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_15:
+        case 15:
             _name = "KUNAI";
             _code_text = "M";
             _shop_price = 300;
             break;
         
-        case UnknownEnum.Value_16:
+        case 16:
             _name = "CROWN";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_17:
+        case 17:
             _name = "BOARD 3";
             _code_text = "M";
             _shop_price = 330;
             break;
         
-        case UnknownEnum.Value_18:
+        case 18:
             _name = "BONE 4";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_19:
+        case 19:
             _name = "BONE 5";
             _code_text = "M";
             _shop_price = 350;
             break;
         
-        case UnknownEnum.Value_20:
+        case 20:
             _name = "CRATE 3";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_22:
+        case 22:
             _name = "SPIKE 1";
             _code_text = "M";
             _shop_price = 380;
             break;
         
-        case UnknownEnum.Value_23:
+        case 23:
             _name = "SPIKE 2";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_24:
+        case 24:
             _name = "PORTRAIT";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_25:
+        case 25:
             _name = "BUCKET";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_26:
+        case 26:
             _name = "PILLORY";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_21:
+        case 21:
             _name = "CHAINS 1";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_27:
+        case 27:
             _name = "CHAINS 2";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_28:
+        case 28:
             _name = "FLAG";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_29:
+        case 29:
             _name = "SACK 1";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_30:
+        case 30:
             _name = "ROCK 1";
             _code_text = "M";
             _shop_price = 250;
             break;
         
-        case UnknownEnum.Value_31:
+        case 31:
             _name = "ROCK 2";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_32:
+        case 32:
             _name = "HAY 1";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_33:
+        case 33:
             _name = "HAY 2";
             _code_text = "M";
             _shop_price = 230;
             break;
         
-        case UnknownEnum.Value_34:
+        case 34:
             _name = "EGG 1";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_35:
+        case 35:
             _name = "EGG 2";
             _code_text = "M";
             break;
         
-        case UnknownEnum.Value_36:
+        case 36:
             _name = "LAMP";
             _code_text = "M";
             _shop_price = 370;
             break;
         
-        case UnknownEnum.Value_38:
+        case 38:
             _name = "BEACON";
             _code_text = "M";
             _shop_price = 450;
             break;
         
-        case UnknownEnum.Value_47:
+        case 47:
             _name = "PLAQUE";
             _code_text = "M";
             break;
@@ -1598,7 +1636,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (button_num)
             {
                 case 0:
@@ -1616,7 +1654,7 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (button_num)
             {
                 case 0:
@@ -1634,7 +1672,7 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (button_num)
             {
                 case 0:
@@ -1652,7 +1690,7 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (button_num)
             {
                 case 0:
@@ -1670,7 +1708,7 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (button_num)
             {
                 case 0:
@@ -1688,7 +1726,7 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (button_num)
             {
                 case 0:
@@ -1706,7 +1744,7 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (button_num)
             {
                 case 0:
@@ -1724,7 +1762,7 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (button_num)
             {
                 case 0:
@@ -1742,7 +1780,7 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (button_num)
             {
                 case 0:
@@ -1760,7 +1798,7 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (button_num)
             {
                 case 0:
@@ -1786,7 +1824,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (button_num)
             {
                 case 0:
@@ -1804,7 +1842,7 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (button_num)
             {
                 case 0:
@@ -1822,7 +1860,7 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (button_num)
             {
                 case 0:
@@ -1840,7 +1878,7 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (button_num)
             {
                 case 0:
@@ -1858,7 +1896,7 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (button_num)
             {
                 case 0:
@@ -1876,7 +1914,7 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (button_num)
             {
                 case 0:
@@ -1894,7 +1932,7 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (button_num)
             {
                 case 0:
@@ -1912,7 +1950,7 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (button_num)
             {
                 case 0:
@@ -1930,7 +1968,7 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (button_num)
             {
                 case 0:
@@ -1948,7 +1986,7 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (button_num)
             {
                 case 0:
@@ -1974,7 +2012,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             switch (button_num)
             {
                 case 0:
@@ -1988,7 +2026,7 @@
             
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             switch (button_num)
             {
                 case 0:
@@ -2002,7 +2040,7 @@
             
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             switch (button_num)
             {
                 case 0:
@@ -2016,7 +2054,7 @@
             
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             switch (button_num)
             {
                 case 0:
@@ -2030,7 +2068,7 @@
             
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             switch (button_num)
             {
                 case 0:
@@ -2044,7 +2082,7 @@
             
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             switch (button_num)
             {
                 case 0:
@@ -2058,7 +2096,7 @@
             
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             switch (button_num)
             {
                 case 0:
@@ -2072,7 +2110,7 @@
             
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             switch (button_num)
             {
                 case 0:
@@ -2086,7 +2124,7 @@
             
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             switch (button_num)
             {
                 case 0:
@@ -2100,7 +2138,7 @@
             
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             switch (button_num)
             {
                 case 0:
@@ -2116,50 +2154,14 @@
     }
 }
 
-enum UnknownEnum
+function scr_create_xl_slot_category_button()
 {
-    Value_0,
-    Value_1,
-    Value_2,
-    Value_3,
-    Value_4,
-    Value_5,
-    Value_6,
-    Value_7,
-    Value_8,
-    Value_9,
-    Value_10,
-    Value_11,
-    Value_12,
-    Value_13,
-    Value_14,
-    Value_15,
-    Value_16,
-    Value_17,
-    Value_18,
-    Value_19,
-    Value_20,
-    Value_21,
-    Value_22,
-    Value_23,
-    Value_24,
-    Value_25,
-    Value_26,
-    Value_27,
-    Value_28,
-    Value_29,
-    Value_30,
-    Value_31,
-    Value_32,
-    Value_33,
-    Value_34,
-    Value_35,
-    Value_36,
-    Value_37,
-    Value_38,
-    Value_39,
-    Value_40,
-    Value_41,
-    Value_42,
-    Value_47 = 47
+    scr_create_button_slot(interact_type, button_num);
+    
+    switch (button_num)
+    {
+        case 0:
+            button_text = "BREED";
+            break;
+    }
 }
