--- gml_138/gml_GlobalScript_s_textbox_function.gml	2026-09-02 08:45:16.167546400 +0200
+++ gml/gml_GlobalScript_s_textbox_function.gml	2026-09-24 15:31:46.000000000 +0200
@@ -225,6 +225,9 @@
         if (text_data.event_after != -1)
             scr_add_event(text_data.event_after, false);
         
+        if (variable_struct_exists(text_data, "gnx_next") && text_data.gnx_next != undefined)
+            scr_gnx_fire_event(text_data.gnx_next);
+        
         instance_destroy();
         
         with (source_id)
