--- gml_138/gml_Object_obj_tent_Draw_0.gml	2026-09-02 08:45:16.161545900 +0200
+++ gml/gml_Object_obj_tent_Draw_0.gml	2026-10-04 17:12:20.000000000 +0200
@@ -1 +1,8 @@
+// POPRAWKA: macki stracily culling w 1.4.6 (usuniety ze skryptu, nie dodany tutaj)
+var _vy = camera_get_view_y(view_camera[0]);
+var _vh = camera_get_view_height(view_camera[0]);
+
+if (y < (_vy - 200) || y > (_vy + _vh + 200))
+    exit;
+
 scr_tent_draw();
