--- gml_138/gml_GlobalScript_s_raid_cal_stat.gml	2026-09-02 08:45:16.522590500 +0200
+++ gml/gml_GlobalScript_s_raid_cal_stat.gml	2026-09-24 15:31:46.000000000 +0200
@@ -282,8 +282,9 @@
             {
                 var _a = irandom_range(0, _list_num - 1);
                 var _unit_data = _data.unit_list[_a];
+                var _gnx_cls_mul = min((_unit_data.class - UnknownEnum.Value_9) + 1, 5);
                 
-                if (_unit_data.class >= UnknownEnum.Value_9 && _list_num > 1 && irandom_range(1, 100) <= (((_unit_data.class - UnknownEnum.Value_9) + 1) * 20 * ((array_length(global.val.cage) - 1) * 0.2)) && array_length(_unit_list2) > 1)
+                if (_unit_data.class >= UnknownEnum.Value_9 && _list_num > 1 && irandom_range(1, 100) <= (_gnx_cls_mul * 20 * ((array_length(global.val.cage) - 1) * 0.2)) && array_length(_unit_list2) > 1)
                 {
                     var _aa = _a;
                     
