--- gml_138/gml_Object_obj_button_Step_0.gml	2026-09-02 08:45:16.360549600 +0200
+++ gml/gml_Object_obj_button_Step_0.gml	2026-10-05 10:01:02.000000000 +0200
@@ -1,3 +1,12 @@
+if (!gui)
+{
+    var _vy = camera_get_view_y(view_camera[0]);
+    var _vh = camera_get_view_height(view_camera[0]);
+    
+    if (y < (_vy - 400) || y > (_vy + _vh + 400))
+        exit;
+}
+
 if (scr_button_step != -1)
     script_execute(scr_button_step);
 
