--- gml_138/gml_GlobalScript_s_raid_draw.gml	2026-09-02 08:45:16.419590000 +0200
+++ gml/gml_GlobalScript_s_raid_draw.gml	2026-09-24 15:31:46.000000000 +0200
@@ -10,7 +10,7 @@
         
         switch (global.val.language)
         {
-            case UnknownEnum.Value_0:
+            case 0:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "INVENTORY");
                 draw_text(30 + _lang_pos[0], 47 + _lang_pos[1], "NEST");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "FORMATION");
@@ -19,7 +19,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "CAGE");
                 break;
             
-            case UnknownEnum.Value_9:
+            case 9:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "背包");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "巢穴");
                 draw_text(391 + _lang_pos[0], 43 + _lang_pos[1], "編隊");
@@ -28,7 +28,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "籠子");
                 break;
             
-            case UnknownEnum.Value_2:
+            case 2:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "背包");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "巢穴");
                 draw_text(391 + _lang_pos[0], 43 + _lang_pos[1], "编队");
@@ -37,7 +37,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "笼子");
                 break;
             
-            case UnknownEnum.Value_1:
+            case 1:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "持ち物");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "巣");
                 draw_text(391 + _lang_pos[0], 43 + _lang_pos[1], "陣形");
@@ -46,7 +46,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "檻");
                 break;
             
-            case UnknownEnum.Value_3:
+            case 3:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "소지품");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "둥지");
                 draw_text(391 + _lang_pos[0], 43 + _lang_pos[1], "대형");
@@ -55,7 +55,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "우리");
                 break;
             
-            case UnknownEnum.Value_7:
+            case 7:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "ИНВЕНТАРЬ");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "ГНЕЗД");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "СТРОЙ");
@@ -64,7 +64,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "КЛЕТКА");
                 break;
             
-            case UnknownEnum.Value_4:
+            case 4:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "INVENTARIO");
                 draw_text(30 + _lang_pos[0], 47 + _lang_pos[1], "NIDO");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "FORMACIÓN");
@@ -73,7 +73,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "JAULA");
                 break;
             
-            case UnknownEnum.Value_6:
+            case 6:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "INVENTAIRE");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "NID");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "FORMATION");
@@ -82,7 +82,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "CAGE");
                 break;
             
-            case UnknownEnum.Value_8:
+            case 8:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "INVENTAR");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "NEST");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "FORMATION");
@@ -91,7 +91,7 @@
                 draw_text(424 + _lang_pos[0], 165 + _lang_pos[1], "KÄFIG");
                 break;
             
-            case UnknownEnum.Value_5:
+            case 5:
                 draw_text(41 + _lang_pos[0], 158 + _lang_pos[1], "INVENTÁRIO");
                 draw_text(31 + _lang_pos[0], 47 + _lang_pos[1], "NINHO");
                 draw_text(390 + _lang_pos[0], 43 + _lang_pos[1], "FORMAÇÃO");
@@ -102,32 +102,37 @@
         }
         
         draw_set_font(fnt_base_font_eng);
-        draw_sprite(spr_map, 0, 173, 39);
+        
+        if (global.gnx_multimap_enabled && global.gnx_map_sprite != -1 && sprite_exists(global.gnx_map_sprite))
+            draw_sprite(global.gnx_map_sprite, 0, 173, 39);
+        else
+            draw_sprite(spr_map, 0, 173, 39);
+        
         var _black = false;
         
         switch (raid_data.raid_state)
         {
-            case UnknownEnum.Value_0:
+            case 0:
                 scr_draw_stage_info();
                 break;
             
-            case UnknownEnum.Value_1:
+            case 1:
                 scr_draw_stage_info();
                 scr_draw_stage_icon();
                 _black = true;
                 break;
             
-            case UnknownEnum.Value_2:
+            case 2:
                 scr_draw_stage_info();
                 scr_draw_stage_icon();
                 _black = true;
                 break;
             
-            case UnknownEnum.Value_3:
+            case 3:
                 scr_draw_raid_trade();
                 break;
             
-            case UnknownEnum.Value_4:
+            case 4:
                 scr_draw_raid_shop();
                 break;
         }
@@ -160,43 +165,43 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             _row_text = ["FRONT", "BACK"];
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             _row_text = ["前列", "後列"];
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             _row_text = ["前列", "后列"];
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             _row_text = ["前列", "後列"];
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             _row_text = ["전열", "후열"];
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             _row_text = ["ФРОНТ", "ТЫЛ"];
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             _row_text = ["FRENTE", "DETRÁS"];
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             _row_text = ["AVANT", "FOND"];
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             _row_text = ["VORNE", "HINTEN"];
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             _row_text = ["FRENTE", "DEFESA"];
             break;
     }
@@ -204,7 +209,7 @@
     var _row_y = [64, 119];
     var _icon_y = 0;
     
-    if (global.val.language == UnknownEnum.Value_9 || global.val.language == UnknownEnum.Value_2 || global.val.language == UnknownEnum.Value_1 || global.val.language == UnknownEnum.Value_3)
+    if (global.val.language == 9 || global.val.language == 2 || global.val.language == 1 || global.val.language == 3)
         _icon_y = 1;
     
     var _line_c = line_c;
@@ -222,7 +227,7 @@
         draw_set_color(_text_c);
         var _a;
         
-        if (raid_data.raid_state == UnknownEnum.Value_1)
+        if (raid_data.raid_state == 1)
             _a = max(sign(raid_data.ap[i]), floor(raid_data.ap[i]));
         else
             _a = max(sign(raid_data.ap[i]), floor(raid_data.ap[i] * _md.md_rate));
@@ -241,7 +246,7 @@
     
     switch (global.val.language)
     {
-        case UnknownEnum.Value_0:
+        case 0:
             draw_text(183 + _lang_pos[0], 134 + _lang_pos[1], "FRONT");
             draw_text(185 + _lang_pos[0], 145 + _lang_pos[1], "BACK");
             draw_text(182 + _lang_pos[0], 156 + _lang_pos[1], "CAPTURE");
@@ -249,7 +254,7 @@
             draw_text(283 + _lang_pos[0], 203 + _lang_pos[1], "TIME");
             break;
         
-        case UnknownEnum.Value_9:
+        case 9:
             draw_text(183 + _lang_pos[0], 133 + _lang_pos[1], "前列 ");
             draw_text(183 + _lang_pos[0], 144 + _lang_pos[1], "後列");
             draw_text(183 + _lang_pos[0], 155 + _lang_pos[1], "捕獲");
@@ -257,7 +262,7 @@
             draw_text(282 + _lang_pos[0], 202 + _lang_pos[1], "時間");
             break;
         
-        case UnknownEnum.Value_2:
+        case 2:
             draw_text(183 + _lang_pos[0], 133 + _lang_pos[1], "前列 ");
             draw_text(183 + _lang_pos[0], 144 + _lang_pos[1], "后列");
             draw_text(183 + _lang_pos[0], 155 + _lang_pos[1], "捕获");
@@ -265,7 +270,7 @@
             draw_text(282 + _lang_pos[0], 202 + _lang_pos[1], "时间");
             break;
         
-        case UnknownEnum.Value_1:
+        case 1:
             draw_text(183 + _lang_pos[0], 133 + _lang_pos[1], "前列");
             draw_text(183 + _lang_pos[0], 144 + _lang_pos[1], "後列");
             draw_text(183 + _lang_pos[0], 155 + _lang_pos[1], "捕獲");
@@ -273,7 +278,7 @@
             draw_text(282 + _lang_pos[0], 202 + _lang_pos[1], "時間");
             break;
         
-        case UnknownEnum.Value_3:
+        case 3:
             draw_text(183 + _lang_pos[0], 133 + _lang_pos[1], "전열 ");
             draw_text(183 + _lang_pos[0], 144 + _lang_pos[1], "후열");
             draw_text(183 + _lang_pos[0], 155 + _lang_pos[1], "생포");
@@ -281,7 +286,7 @@
             draw_text(282 + _lang_pos[0], 202 + _lang_pos[1], "시간");
             break;
         
-        case UnknownEnum.Value_7:
+        case 7:
             draw_text(182 + _lang_pos[0], 134 + _lang_pos[1], "ФРОНТ");
             draw_text(185 + _lang_pos[0], 145 + _lang_pos[1], "ТЫЛ");
             draw_text(182 + _lang_pos[0], 156 + _lang_pos[1], "ЗАХВАТ");
@@ -289,7 +294,7 @@
             draw_text(281 + _lang_pos[0], 203 + _lang_pos[1], "ВРЕМЯ");
             break;
         
-        case UnknownEnum.Value_4:
+        case 4:
             draw_text(181 + _lang_pos[0], 134 + _lang_pos[1], "FRENTE");
             draw_text(181 + _lang_pos[0], 145 + _lang_pos[1], "DETRÁS");
             draw_text(181 + _lang_pos[0], 156 + _lang_pos[1], "CAPTURA");
@@ -297,7 +302,7 @@
             draw_text(279 + _lang_pos[0], 203 + _lang_pos[1], "TIEMPO");
             break;
         
-        case UnknownEnum.Value_6:
+        case 6:
             draw_text(183 + _lang_pos[0], 134 + _lang_pos[1], "AVANT");
             draw_text(185 + _lang_pos[0], 145 + _lang_pos[1], "FOND");
             draw_text(182 + _lang_pos[0], 156 + _lang_pos[1], "CAPTURE");
@@ -305,7 +310,7 @@
             draw_text(280 + _lang_pos[0], 203 + _lang_pos[1], "TEMPS");
             break;
         
-        case UnknownEnum.Value_8:
+        case 8:
             draw_text(183 + _lang_pos[0], 134 + _lang_pos[1], "VORNE");
             draw_text(181 + _lang_pos[0], 145 + _lang_pos[1], "HINTEN");
             draw_text(181 + _lang_pos[0], 156 + _lang_pos[1], "EINFANGEN");
@@ -313,7 +318,7 @@
             draw_text(283 + _lang_pos[0], 203 + _lang_pos[1], "ZEIT");
             break;
         
-        case UnknownEnum.Value_5:
+        case 5:
             draw_text(181 + _lang_pos[0], 134 + _lang_pos[1], "FRENTE");
             draw_text(181 + _lang_pos[0], 145 + _lang_pos[1], "DEFESA");
             draw_text(181 + _lang_pos[0], 156 + _lang_pos[1], "CAPTURAR");
@@ -333,7 +338,7 @@
         var _data = global.val.stage_info[raid_data.stage].info[_stage_lvl];
         var _player_hp, _player_ap;
         
-        if (raid_data.raid_state != UnknownEnum.Value_1)
+        if (raid_data.raid_state != 1)
         {
             var _md = scr_calculate_mood_rate();
             _player_hp = floor(obj_np.obj_raid.raid_data.ap[0] * _md.md_rate);
@@ -402,8 +407,8 @@
         
         if (_player_hp >= _enemy_hp && _player_ap >= _enemy_ap)
         {
-            _cap_mul = scr_check_skill_data(UnknownEnum.Value_5);
-            _rev_mul = scr_check_skill_data(UnknownEnum.Value_1);
+            _cap_mul = scr_check_skill_data(5);
+            _rev_mul = scr_check_skill_data(1);
             var _over_diff = (_player_hp + _player_ap) / (_enemy_hp + _enemy_ap);
             var _list_num = array_length(_data.unit_list);
             var _max_lvl = 0;
@@ -412,17 +417,17 @@
             {
                 if (_data.unit_list[i].lvl > _max_lvl)
                 {
-                    if (_data.unit_list[i].class == UnknownEnum.Value_8)
+                    if (_data.unit_list[i].class == 8)
                         _max_lvl = 8;
                     else
                         _max_lvl = _data.unit_list[i].lvl;
                 }
                 
-                if (global.unlock.boss[UnknownEnum.Value_3] == 0 && _data.unit_list[i].class == UnknownEnum.Value_12)
+                if (global.unlock.boss[3] == 0 && _data.unit_list[i].class == 12)
                     _dd = true;
             }
             
-            if (raid_data.stage == UnknownEnum.Value_4 && _stage_lvl == 1)
+            if (raid_data.stage == 4 && _stage_lvl == 1)
                 _dd = true;
             
             _num = ((_over_diff - 1) / (1 + (_max_lvl * 0.3))) * 100;
@@ -461,7 +466,7 @@
         draw_text(241, 156, "%");
         draw_text(273, 156, "%");
         
-        if (raid_data.raid_state == UnknownEnum.Value_0)
+        if (raid_data.raid_state == 0)
         {
             _x = 285;
             _y = 129;
@@ -478,16 +483,16 @@
 {
     draw_sprite(spr_button_stage, 0, 240, 92);
     
-    if (global.val.stage_info[UnknownEnum.Value_1] != -1)
+    if (global.val.stage_info[1] != -1)
         draw_sprite(spr_button_stage, 1, 188, 108);
     
-    if (global.val.stage_info[UnknownEnum.Value_2] != -1)
+    if (global.val.stage_info[2] != -1)
         draw_sprite(spr_button_stage, 2, 266, 50);
     
-    if (global.val.stage_info[UnknownEnum.Value_3] != -1)
+    if (global.val.stage_info[3] != -1)
         draw_sprite(spr_button_stage, 3, 272, 103);
     
-    if (global.val.stage_info[UnknownEnum.Value_4] != -1)
+    if (global.val.stage_info[4] != -1)
         draw_sprite(spr_button_stage, 4, 294, 80);
     
     draw_sprite(spr_raid_shop_button, 0, 233, 50);
@@ -496,18 +501,3 @@
     if (instance_exists(obj_carriage))
         draw_sprite(spr_carriage, image_index, obj_carriage.x, obj_carriage.y);
 }
-
-enum UnknownEnum
-{
-    Value_0,
-    Value_1,
-    Value_2,
-    Value_3,
-    Value_4,
-    Value_5,
-    Value_6,
-    Value_7,
-    Value_8,
-    Value_9,
-    Value_12 = 12
-}
