--- gml_138/gml_GlobalScript_s_text_draw.gml	2026-09-02 08:45:16.363596700 +0200
+++ gml/gml_GlobalScript_s_text_draw.gml	2026-09-24 15:31:46.000000000 +0200
@@ -26,7 +26,12 @@
 {
     scr_draw_backdrop();
     draw_sprite(spr_raid_screen, 0, 170, 36);
-    draw_sprite(spr_portrait, portrait_index, 173, 39);
+    
+    if (portrait_index >= 100 && sprite_exists(portrait_index))
+        draw_sprite(portrait_index, 0, 173, 39);
+    else
+        draw_sprite(spr_portrait, portrait_index, 173, 39);
+    
     var _lang_pos = scr_change_lang();
     var _y = 0;
     draw_set_halign(fa_center);
@@ -967,7 +972,7 @@
                             break;
                         
                         case UnknownEnum.Value_1:
-                            _text_1 = "Monster haben eine Chance zu sterben.";
+                            _text_1 = "Monster haben eine Chance nicht zu sterben.";
                             _text_2 = "Erhöhe 4% pro Stapel.";
                             break;
                         
