--- gml_138/gml_GlobalScript_s_unit_head.gml	2026-09-02 08:45:16.156546900 +0200
+++ gml/gml_GlobalScript_s_unit_head.gml	2026-10-04 17:12:20.000000000 +0200
@@ -77,7 +77,8 @@
         {
             with (obj_slot)
             {
-                if ((slot_data.slot_type == UnknownEnum.Value_0 || slot_data.slot_type == UnknownEnum.Value_1 || slot_data.slot_type == UnknownEnum.Value_2) && slot_data.anim_struct != -1 && (unit_data != -1 || slot_data.load))
+                // POPRAWKA: ikony postaci takze nad celami XL
+                if ((slot_data.slot_type == UnknownEnum.Value_0 || slot_data.slot_type == UnknownEnum.Value_1 || slot_data.slot_type == UnknownEnum.Value_2 || slot_data.slot_type == 6) && slot_data.anim_struct != -1 && (unit_data != -1 || slot_data.load))
                 {
                     var _x = floor(image_xscale * 0.5);
                     var _head = scr_create_button(_x, -96, -1001, spr_unit_icon_head, UnknownEnum.Value_20, id, scr_create_unit_head, -1, scr_draw_unit_head, 0, 0);
